src/import/ImportAUP.cpp

```diff
@@ -645,112 +645,101 @@ bool AUPImportFileHandle::HandleProject(XMLTagHandler *&handler)
 
    int requiredTags = 0;
 
-   while (*mAttrs)
+   for (auto pair : mAttrs)
    {
-      const wxChar *attr = *mAttrs++;
-      const wxChar *value = *mAttrs++;
-      double dValue;
-
-      if (!value)
-      {
-         break;
-      }
+      auto attr = pair.first;
+      auto value = pair.second;
```

Comment:
This line made me look here and check the rules for copying of unions https://en.cppreference.com/w/cpp/language/union